app/models/spree/stock_movement.rb

```diff
@@ -2,14 +2,11 @@
 
 module Spree
   class StockMovement < ApplicationRecord
-    self.belongs_to_required_by_default = false
-
     belongs_to :stock_item, class_name: 'Spree::StockItem'
-    belongs_to :originator, polymorphic: true
+    belongs_to :originator, polymorphic: true, optional: true
```

Comment:
I am not too familiar with the stock system but I feel like this shouldn't be optional @openfoodfoundation/core-devs ?

Reply:
Without the `optional: true`, around 10 specs are in error, all related to originator.

Reply:
Maybe the specs need to set the originator? Is it just incomplete test data?

Reply:
Hello @mkllnk  ,

I have no idea if the `originator` nulleable is a good thing or not.
I have watched a bit one example and cannot make my mind.
I also have no data for `stock movement` locally.

But, I have looked at the `adjustment` model that belongs too to `originator` (There are only 2) 
(Btw for now with the `self.belongs_to_required_by_default = false`)

There are various scope defined, amongst them this one:
`scope :admin,          -> { where(originator_type: nil) }`
Which indicates that for the `adjustment` model, the `originator` may be null.
So my guess is that nulleable is a good choice.

May be someone can help us with the data model and what are the exact intentions of this particular model ?

Reply:
I looked at the production server, and we have some stock movement without originator :
```
sql=`select count(*) from spree_stock_movements where originator_id is null;'

ansible all_prod -u openfoodnetwork -a "psql -h localhost openfoodnetwork ofn_user -c '$sql'"
openfoodnetwork.org.au | CHANGED | rc=0 >>
 count 
-------
     0
(1 row)
openfoodnetwork.org.nz | CHANGED | rc=0 >>
 count 
-------
     0
(1 row)
openfoodnetwork.ca | CHANGED | rc=0 >>
 count 
-------
     0
(1 row)
openfoodnetwork.org.uk | CHANGED | rc=0 >>
 count 
-------
     3
(1 row)
openfoodnetwork.ie | CHANGED | rc=0 >>
 count 
-------
     0
(1 row)
openfoodnetwork.net | CHANGED | rc=0 >>
 count 
-------
     3
(1 row)
app.katuma.org | CHANGED | rc=0 >>
 count 
-------
    26
(1 row)
csicsoka.org | CHANGED | rc=0 >>
 count 
-------
     0
(1 row)
openfoodnetwork.de | CHANGED | rc=0 >>
 count 
-------
     0
(1 row)
openfoodnetwork.in | CHANGED | rc=0 >>
 count 
-------
     0
(1 row)
openfoodnetwork.be | CHANGED | rc=0 >>
 count 
-------
     1
(1 row)
coopcircuits.fr | CHANGED | rc=0 >>
 count 
-------
    16
(1 row)
```
So maybe we want to keep it as it is ? If I remember correctly we are trying to not change inventory, so that's another reason not to change this ?

I am not sure if an admin user can do manual stock movement ? but from what I can infer from the data, all stock movement orginator are `Spree::Shipment`. In the case of an `adjustment` an adjustment with no originator is manual adjustment, ie made by an admin. So it would be fair to assume the same here ?

@mkllnk what do you think ?

Reply:
This pull request is addressing style rules. So let's keep the logic as it is and go ahead with this.